shell.nix

```diff
@@ -0,0 +1,23 @@
+{ pkgs ? import <nixpkgs> {
+    overlays = [
+      (import (fetchTarball
+        "https://github.com/nix-community/fenix/archive/main.tar.gz"))
+    ];
+  }
+}:
+
+pkgs.mkShell {
+  packages = with pkgs; [
+    rustc
+    cargo
+    rust-analyzer
+    gcc
+    gtk3
+    pkg-config
+    rustfmt
+    clippy
+  ];
+
+
+  RUST_SRC_PATH = "${pkgs.rust.packages.stable.rustPlatform.rustLibSrc}";
```

Comment:
This points to the normal rustc that's packaged in nixpkgs, right?

Can this be replaced with the one that fenix provides?

Reply:
Done it. and the newlines are part of nixfmt, and get readded on reformat